swift/extractor/main.cpp

```diff
@@ -96,6 +104,23 @@ static void checkWhetherToRunUnderTool(int argc, char* const* argv) {
   execvp(args[0], args.data());
 }
 
+// Creates a target file that should store per-invocation info, e.g. compilation args,
+// compilations, diagnostics, etc.
+codeql::TargetFile invocationTargetFile(codeql::SwiftExtractorConfiguration& configuration) {
+  auto hasher = picosha2::hash256_one_by_one();
+  for (auto& option : configuration.frontendOptions) {
+    hasher.process(option.c_str(), option.c_str() + option.size() + 1);
+  }
+  hasher.finish();
+  auto target = "invocation-"s + get_hash_hex_string(hasher);
+  auto maybeFile = codeql::createTargetTrapFile(configuration, target);
+  if (!maybeFile) {
+    std::cerr << "Cannot create invocation trap file: " << target << "\n";
+    abort();
+  }
```

Comment:
what if for some reason we are calling the swift frontend with exactly the same arguments? Should we really abort in such a case? Also, now that I think about it hashing the command line flags might miss changes to the files provided to the call...

As there is no need to reference this output trap file outside of this very call of the extractor, maybe it'd be better to just use a random new file name every time?

Reply:
Yeah, you are right, some random name would work better.

Reply:
as discussed on chat, probably timestamp + getpid is a good fit for the trap target. Also, maybe use `invocations` as a directory containing the traps rather than as a filename prefix?